openghg/retrieve/_search.py

```diff
@@ -605,11 +605,20 @@ def _base_search(**kwargs: Any) -> SearchResults:
             logger.exception(msg=error_msg)
             raise ValueError(error_msg)
 
-        # Here we create a dictionary of the metadata keyed by the Datasource UUID
-        # we'll create a pandas DataFrame out of this in the SearchResult object
-        # for better printing / searching within a notebook
+        # Due to versions of OpenGHG < 0.6.0 not storing the object store
+        # in the metadata we nede to pass that information in here for backwards
+        # compatibility.
         metadata = {r["uuid"]: r for r in metastore_records}
 
+        updated_metadata = {}
+        store_data = {"object_store": bucket}
```

Comment:
Is this adding in the "object_store" key for every object store this interacts with then? I'm just thinking if the "object_store" key is already present we wouldn't want this to be silently replaced here without us realising so may want to add a check if this is present before replacing it.

Reply:
Yeah I'm going to add a check in there and make sure the values don't differ.